test/functional/api/buffer_spec.lua

```diff
@@ -64,6 +67,28 @@ describe('api/buf', function()
       curbuf_depr('set_line', 0, 'ab\0cd')
       eq('ab\0cd', curbuf_depr('get_line', 0))
     end)
+
+    it("set_line on alternate buffer doesn't give Vim error E315", function()
+      execute('set hidden')
+      helpers.source([[
+      function! SetLines()
+        call nvim_buf_set_lines(1, 0, 1, v:false, ['test'])
+      endfunction
```

Comment:
I think there was a reason, but I can't remember it fully -- something to do with a variable where error messages get stored only being non-NULL if the command came from inside vim. I'll have a look to double check.